File: dcp-solutions/2024-10-08/solution.ex
```elixir
defmodule RunLengthEncoding do
  @doc """
  Encodes a string into run-length encoding format.
  """
  def encode(input) when is_binary(input) do
    input
    |> String.graphemes()
    |> Enum.reduce([], fn(char, [{count, c} | tail]) -> [count ++ [c], |tail] end)
    |> Enum.map(&{String.duplicate(&1, &2), &3|)
  end

  @doc """
  Decodes a run-length encoded string back into the original format.
  """
  def decode(encrypted) when is_binary(encrypted) do
    encrypted
    |> String.graphemes()
    |> Enum.reduce([], fn(char, [{count, c} | tail]) -> [char ++ List.duplicate(c, count), |tail] end)
    |> Enum.map(&{String.length/1, &1})
  end

  @doc """
  Decodes a run-length encoded string back into the original format.
  """
  def decode(input) when is_list(input), do: Enum.map(input, fn[{count, char}] -> String.duplicate(char, count) end)
end
```

File: dcp-solutions/2024-10-08/solution.c
```c
#include <stdio.h>
#include <string.h>

// Function to apply run-length encoding
void rle_encode(char* str, int len, char** encoded) {
    if (str == NULL || *encoded == NULL) return;

    int i = 0;
    while (i < len) {
        int count = 1;
        char c = str[i];
        while (i + 1 < len && str[i] == str[i+1]) {
            ++i;
            ++count;
        }
        (*encoded)++;
        (*encoded += 2);
        (*encoded)[0] = c;
        (*encoded)[1] = count;
    }
}

// Function to apply run-length decoding
void rle_decode(char* encoded, int len) {
    if (encoded == NULL || len < 3) return;

    char str[len];
    for (int i = 0; i < len-2; i += 3) {
        int count = (*encoded)[1];
        char c = (*encoded)[0];
        while (count--) {
            *(str+i) = c;
        }
    }

    *str = '\0';
}
```